lib/bucky/core/test_core/test_manager.rb

```diff
@@ -8,7 +8,51 @@
 module Bucky
   module Core
     module TestCore
+      module ParallelHelper
+        parent_pid = Process.pid
+
+        # Terminate parent and child process when getting interrupt signal
+        Signal.trap('INT') do
+          Process.kill('TERM', -1 * parent_pid)
+        end
+
+        private
+
+        def parallel_new_worker_each(data_set, max_processes, &block)
+          # Max parallel workers number
+          available_workers = max_processes
+
+          # If child process dead, available workers increase
+          Signal.trap('CLD') { available_workers += 1 }
+
+          data_set.each do |data|
+            # Wait until worker is available
+            Process.wait unless available_workers.positive?
+            # Workers decrease when start working
+            available_workers -= 1
+            fork { block.call(data) }
+          end
+          Process.waitall
+        end
+
+        def parallel_distribute_into_workers(data_set, max_processes, &block)
+          # For checking on linkstatus
+          divisor_of_data = max_processes
+          num_of_works_in_pre_worker = (data_set.length.to_f / divisor_of_data.to_f).ceil
+          # Slice data_set into few parts that depends on workers
+          data_set.each_slice(num_of_works_in_pre_worker) do |data_for_pre_worker|
```

Comment:
I think there are times when this implementation doesn't work fine.
```
max =4
data_set = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

(data_set.length.to_f / 4.to_f).ceil
# => 3

data_set.each_slice(3) {|d| p d }
# =>[1, 2, 3]
# =>[4, 5, 6]
# =>[7, 8, 9]
# =>[10]

# Works fine
# =>[1, 2, 3]
# =>[4, 5, 6]
# =>[7, 8]
# =>[9, 10]
```

How about implementation like this?
```
max = 4
data_set = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

devided_data_set = Array.new
max.times {|i| devided_data_set << [] }
data_index = 0
data_set.each do |d|
  data_index = 0 if data_index >= max
  devided_data_set[data_index] << d
  data_index +=1
end
p devided_data_set

# => [[1, 5, 9], [2, 6, 10], [3, 7], [4, 8]]
```

Reply:
Thanks of your advise.
I want to change it like bellow.
```
max =4
data_set = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

data_set_grouped = data_set.group_by.with_index { |elem,index| index%max }
p data_set_grouped.values

# =>  [[1, 5, 9], [2, 6, 10], [3, 7], [4, 8]]
```